### ISSUE TITLE
URL encoding for `Authority` should use UTF-8

### ISSUE THREAD

**Issue by edsko** [kazu-yamamoto/http2-tls#7]
In `http2` we have

```haskell
type Authority = ByteString
data ClientConfig = ClientConfig { authority :: Authority, .. }
```

In `http2-tls`, we have instead

```haskell
type HostName = String
defaultClientConfig :: Settings -> HostName -> ClientConfig
defaultClientConfig Settings{..} serverName =
    H2Client.defaultClientConfig { H2Client.authority = C8.pack serverName, .. }
```

That call to `C8.pack` to turn the `String` (`HostName`) into a `ByteString` (`Authority`) is not quite right, I think; [the spec mandates UTF8](https://www.rfc-editor.org/rfc/rfc3986#section-3.2.2). 

I don't have an explicit use case for this, I just noticed it as I was integrating the new `http2-tls`. Changing it is not backwards compatible (it wouldn't have been in PR #6 either). Not urgent, as far as I am concerned.

**Comment by kazu-yamamoto** [comment]
So, should we change the definition as

```
type Authority = HostName
```

in `http2` and bumps the major version?

**Comment by edsko** [comment]
I think that's probably a good idea; I took a look to see if this would result in additional complications, but it seems it's straight-forward, see https://github.com/kazu-yamamoto/http2/pull/105 . 

Probably the Right Thing is to then change `http2-tls` to define `Authority` instead of (in addition to..?) `HostName`, or better yet, re-use the definition from `http2`.

@kazu-yamamoto If you are going to release a new version of `http2`, would you feel comfortable in including https://github.com/kazu-yamamoto/http2/pull/104 also?

**Comment by kazu-yamamoto** [comment]
It's Friday evening now in Japan.
Maybe in the next week.

**Comment by edsko** [comment]
Sure, of course :) Have a good weekend! :)

**Comment by kazu-yamamoto** [comment]
https://github.com/kazu-yamamoto/http2/pull/105 has been merged.
It's `http2-tls`'s turn.

**Comment by kazu-yamamoto** [comment]
@edsko  `master` now uses `http2` v5.1 which is not released yet.
If you want to use `Authority` instead of `HostName`, please send a PR.

**Comment by edsko** [comment]
Coming up! :)

**Comment by edsko** [comment]
https://github.com/kazu-yamamoto/http2-tls/pull/8

**Comment by edsko** [comment]
This was closed in #8.
